### PR TITLE
Add TFRecordSpec support for featran

### PR DESCRIPTION
**Pull request by yonromai**

### CODE REVIEW

**Comment by yonromai** [comment]
@nevillelyh @ravwojdyla PTAL

File: build.sbt [hunk 1]
```diff
@@ -507,7 +508,10 @@ lazy val scioTensorFlow: Project = Project(
   libraryDependencies ++= Seq(
     "org.tensorflow" % "tensorflow" % tensorFlowVersion,
     "org.tensorflow" % "proto" % tensorFlowVersion,
-    "me.lyh" %% "shapeless-datatype-tensorflow" % shapelessDatatypeVersion
+    "me.lyh" %% "shapeless-datatype-tensorflow" % shapelessDatatypeVersion,
+    "com.spotify" %% "featran-core" % featranVersion,
+    "com.spotify" %% "featran-scio" % featranVersion % "test",
```
Comment (nevillelyh) [inline build.sbt]: Any reason not to include these 2 as `"compile"`?

**Comment by yonromai** [comment]
@nevillelyh updated PR

File: scio-tensorflow/src/test/scala/com/spotify/scio/tensorflow/FeatranTFRecordTest.scala [hunk 2]
```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright 2017 Spotify AB.
```
Comment (ravwojdyla) [inline scio-tensorflow/src/test/scala/com/spotify/scio/tensorflow/FeatranTFRecordTest.scala]: 2018

File: scio-tensorflow/src/test/scala/com/spotify/scio/tensorflow/FeatranTFRecordTest.scala [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+      """{"name":"label","kind":"FloatList","tags":{"multispec-id":"1"}}],""" +
+      """"compression":"UNCOMPRESSED"}"""
+
+  "MultiSpecJob" should "work" in {
+    JobTest[MultiSpecJob.type]
+      .args("--input=in.txt", "--output=out")
+      .input(TextIO("in.txt"), in)
+      .output(TextIO("out/train/_tf_record_spec.json"))(_ should containSingleValue(tfRecordMSpec))
+      .output(TextIO("out/test/_tf_record_spec.json"))(_ should containSingleValue(tfRecordMSpec))
+      .output(TFExampleIO("out/train"))(_.count should forAll[Long](c => c > 8500 && c < 9500))
```
Comment (ravwojdyla) [inline scio-tensorflow/src/test/scala/com/spotify/scio/tensorflow/FeatranTFRecordTest.scala]: you can probably use the `+-` right now.

File: scio-tensorflow/src/main/scala/com/spotify/scio/tensorflow/TensorFlowSCollectionFunctions.scala [hunk 4]
```diff
@@ -194,6 +195,75 @@ class TFExampleSCollectionFunctions[T <: Example](val self: SCollection[T]) {
...
+    * @group output
+    */
+  def saveAsTfExampleFile(path: String,
+                          fe: MultiFeatureExtractor[SCollection, _]):
+  (Future[Tap[Example]], Future[Tap[String]]) = {
+    saveAsTfExampleFile(path, fe, Compression.UNCOMPRESSED)
+  }
+
+  /**
+    * Merge each [[Seq]] of [[Example]] and save them as TensorFlow TFRecord files.
```
Comment (ravwojdyla) [inline scio-tensorflow/src/main/scala/com/spotify/scio/tensorflow/TensorFlowSCollectionFunctions.scala]: Invalid comment indent.

File: scio-tensorflow/src/main/scala/com/spotify/scio/tensorflow/TensorFlowSCollectionFunctions.scala [hunk 5]
```diff
@@ -194,6 +195,75 @@ class TFExampleSCollectionFunctions[T <: Example](val self: SCollection[T]) {
...
+
+class SeqTFExampleSCollectionFunctions[T <: Example]
+(@transient val self: SCollection[Seq[T]]) extends Serializable {
+
+  def mergeExamples(e: Seq[Example]): Example = e
+    .foldLeft(Example.newBuilder)((b, i) => b.mergeFrom(i))
+    .build()
+
+  /**
+    * Merge each [[Seq]] of [[Example]] and save them as TensorFlow TFRecord files.
```
Comment (ravwojdyla) [inline scio-tensorflow/src/main/scala/com/spotify/scio/tensorflow/TensorFlowSCollectionFunctions.scala]: invalid comment indent.

**Pull request merged**